### PR TITLE
Fixes warnings & coverity issue on UtilityFunction

### PR DESCRIPTION
**Pull request by hendrik-skubch**

### CODE REVIEW

File: alica_engine/include/engine/constraintmodul/Query.h [hunk 1]
```diff
@@ -169,8 +169,8 @@ bool Query::getSolution(std::shared_ptr<const RunningPlan> rp, std::vector<Resul
         }
         // Again, the queried domain variables should be at the beginning of the domain variable segment
         // So a simple move and resize should do the trick
-        for (int i = 0; i < _queriedDomainVariables.size(); ++i) {
-            for (int j = domOffset; j < _relevantVariables.size(); ++j) {
+        for (int i = 0; i < static_cast<int>(_queriedDomainVariables.size()); ++i) {
```
Comment (ChristofDubs) [inline alica_engine/include/engine/constraintmodul/Query.h]: wouldn't it make more sense to declare `i` as whichever type .size() returns in the first place?

File: alica_engine/src/engine/UtilityFunction.cpp [hunk 2]
```diff
@@ -224,16 +218,13 @@ void UtilityFunction::init(AlicaEngine* ae)
             } else {
                 curPrio = iter->second;
             }
-            TaskRoleStruct* trs = new TaskRoleStruct(taskId, roleId);
-            if (this->priorityMartix.find(trs) == this->priorityMartix.end()) {
-                this->priorityMartix.insert(pair<TaskRoleStruct*, double>(trs, curPrio));
-            }
+            priorityMatrix[TaskRoleStruct(taskId, roleId)] = curPrio;
```
Comment (ChristofDubs) [inline alica_engine/src/engine/UtilityFunction.cpp]: It looks like the behavior change from ignoring duplicate entries to overwriting them. Is that intentional / what are the consequences of this?

File: alica_tests/autogenerated/src/DistBallRobot.cpp [hunk 3]
```diff
@@ -19,31 +21,31 @@ DistBallRobot::DistBallRobot(double weight, string name, long id, vector<long>&
...
-    std::vector<uint8_t> id9(
-            reinterpret_cast<const uint8_t*>(&x9), (reinterpret_cast<const uint8_t*>(&x9) + sizeof(x9)));
+    std::vector<uint8_t> id9(reinterpret_cast<const uint8_t*>(&x9), (reinterpret_cast<const uint8_t*>(&x9) + sizeof(x9)));
     const supplementary::AgentID* agentID9 = this->manager->getIDFromBytes(id9);
 
     const AgentGrp* relevantRobots = ass->getRobotsWorking(this->relevantEntryPoints[0]);
 
     double curPosition;
-    for (int i = 0; i < relevantRobots->size(); ++i) {
+    for (int i = 0; i < static_cast<int>(relevantRobots->size()); ++i) {
```
Comment (ChristofDubs) [inline alica_tests/autogenerated/src/DistBallRobot.cpp]: dito

File: alica_engine/src/engine/UtilityFunction.cpp (hunk 2 above)
Comment (hendrik-skubch) [reply alica_engine/src/engine/UtilityFunction.cpp]: yes that is intentional. Basically, that would indicate bad data. The last value is as good as any.

File: alica_engine/include/engine/constraintmodul/Query.h (hunk 1 above)
Comment (hendrik-skubch) [reply alica_engine/include/engine/constraintmodul/Query.h]: well...there are different opinions about it.
Here's mine:
- size() returns size_t, i.e., uint64_t, and for performance reason i'd rather use a 32 bit counter.
- Imo, using an unsigned value for a size is a bad idea, and should be rectified whenever possible. Google's guideline gives a good explanation:
https://google.github.io/styleguide/cppguide.html#Integer_Types


tl;dr: I stubbornly fight size_t as a container size value.
